src/main/java/servlets/LoginServlet.java

```diff
@@ -26,19 +27,44 @@
 
 @WebServlet("/Login")
 public class LoginServlet extends HttpServlet {
-
+  private static final Logger LOGGER = Logger.getLogger(LoginServlet.class.getName());
   private static final String CONTENT_TYPE_TEXT_HTML = "text/html;";
   private static final String REDIRECT_LINK = "/Login";
+  private final String CLIENT_ID = "client_id=150737768611-svndjtlklolq53g4ass4r3sqal2i31p5.apps.googleusercontent.com";
+  private final String SCOPE = "scope=https://www.googleapis.com/auth/spreadsheets";
+  private final String REDIRECT_URI = "redirect_uri=";
+  private final String OAUTH_LOGIN_URI = "https://accounts.google.com/o/oauth2/v2/auth";
+  private final String RESPONSE_TYPE = "response_type=code";
 
-  // Returns  a URL to either login or logout.
+  // Returns a URL to either login and get OAuth tokens or to logout.
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType(CONTENT_TYPE_TEXT_HTML);
     PrintWriter out = response.getWriter();
     if (UserAuthUtil.isUserLoggedIn()) {
       out.println(UserAuthUtil.getLogoutURL(REDIRECT_LINK));
     } else {
-      out.println(UserAuthUtil.getLoginURL(REDIRECT_LINK));
+      // Send user to OAuth consent page.
+      response.sendRedirect(getOAuthRedirectURL());
+    }
+  }
+
+  /** Build the OAuth consent page redirect URL. */
+  private String getOAuthRedirectURL() {
+    String redirectUri = REDIRECT_URI + getRedirectUri();
+    
+    return String.format("%s?%s&%s&%s&%s", OAUTH_LOGIN_URI, CLIENT_ID,
+      RESPONSE_TYPE, redirectUri, SCOPE); 
+  }
+
+  // Build a valid redirect URI to the OAuthCallbackServlet.
+  private String getRedirectUri() {
+    try {
+      URI domainUri = URI.create(System.getenv().get("DOMAIN"));
+      return domainUri.resolve("/api/oauth/callback/sheets").toString();
+    } catch (NullPointerException e) {
+      LOGGER.severe("The DOMAIN environment variable is not set.");
+      throw e;
```

Comment:
nit: indent +2?

Reply:
This is a correctly sized indent.

Reply:
oops yeah, sorry it looked weird because of my screen size.